### PR TITLE
[WiP] Add org-level config component

### PR DESCRIPTION
**Pull request by purinkle**

### CODE REVIEW

File: app/assets/stylesheets/components/_switch.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
+$switch-height: 32px;
+$switch-padding: 2px;
+$knob-size: $switch-height - ($switch-padding * 2) - 2px;
+$switch-width: 52px;
```
Comment (gylaz) [inline app/assets/stylesheets/components/_switch.scss]: Maybe move these to `app/assets/stylesheets/base/_variables.scss`?

File: spec/factories.rb [hunk 2]
```diff
@@ -91,8 +91,8 @@
   end
 
   factory :owner do
-    github_id
-    name { generate(:github_name) }
+    github_id 1
+    name "TEST_GITHUB_LOGIN"
```
Comment (gylaz) [inline spec/factories.rb]: Why was this change necessary? Both `github_id` and `name` are unique, so creating two `owner` factories in one test would fail.

File: app/assets/javascripts/components/organization_configuration.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    Ajax.updateOwner(this.props.id, Object.assign({}, currentState, options));
+  }
+
+  renderRepoSelect() {
+    if (this.state.enabled) {
+      return (
+        <div className="organization-header-config">
+          <span className="organization-header-label">
+            Use hound.yml from
```
Comment (gylaz) [inline app/assets/javascripts/components/organization_configuration.jsx]: `User .hound.yml from`?

File: app/assets/javascripts/components/organization_configuration.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          <label className="toggle-switch" htmlFor={switchId} />
+        </div>
+
+        {this.renderRepoSelect()}
+      </div>
+    );
+  }
+}
+
+module.exports = OrganizationConfiguration;
```
Comment (gylaz) [inline app/assets/javascripts/components/organization_configuration.jsx]: This is certainly a lot of code for a dropdown and a toggle. Part of the problem I have with JS frameworks. Just my 2 cents.

**Comment by purinkle** [comment]
[Support org-level styleguide.](https://trello.com/c/HY2hoTdq/691-support-org-level-styleguide)

File: app/assets/javascripts/components/organization_configuration.jsx (hunk 4 above)
Comment (purinkle) [inline app/assets/javascripts/components/organization_configuration.jsx]: If the main repo list was not rendered using AJAX but using standard Rails then it would be easier to add these sort of components. For me, it would be better if we could get the whole application to work without JS and then add enhanced functionality with JS. For example, this would just be a small `form` for each organisation that would submit normally. We would then "enhance" the functionality to allow the form to submit via AJAX.
